### PR TITLE
Fix issue #45423 

### PR DESCRIPTION
**Pull request by welovej**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Comment by welovej** [comment]
#45423

File: sdk/dns/Azure.ResourceManager.Dns/src/Customization/DnsRecordData.cs [hunk 1]
```diff
@@ -142,11 +143,25 @@ public string Cname
...
         public IList<DnsTlsaRecordInfo> DnsTlsaRecords { get; }
         /// <summary> The list of NAPTR records in the record set. </summary>
         public IList<DnsNaptrRecordInfo> DnsNaptrRecords { get; }
+        /// <summary> The DnsRecordType in the record set. </summary>
+        public DnsRecordType RecordType
+        {
+            get
+            {
+                var resourceTypeString = base.ResourceType.Type.Split('/').Where(part => !string.IsNullOrEmpty(part)).LastOrDefault();
+                resourceTypeString = resourceTypeString == "AAAA" ? "Aaaa" : resourceTypeString;
```
Comment (live1206) [inline sdk/dns/Azure.ResourceManager.Dns/src/Customization/DnsRecordData.cs:158]: As I can see there are more items in `DnsRecordType`, should we handle more cases here?

File: sdk/dns/Azure.ResourceManager.Dns/src/Generated/Models/DnsCnameRecordInfo.Serialization.cs [hunk 2]
```diff
@@ -13,7 +13,7 @@
 
 namespace Azure.ResourceManager.Dns.Models
 {
-    internal partial class DnsCnameRecordInfo : IUtf8JsonSerializable, IJsonModel<DnsCnameRecordInfo>
+    public partial class DnsCnameRecordInfo : IUtf8JsonSerializable, IJsonModel<DnsCnameRecordInfo>
```
Comment (live1206) [inline sdk/dns/Azure.ResourceManager.Dns/src/Generated/Models/DnsCnameRecordInfo.Serialization.cs:16]: why is this changed to public?

File: sdk/dns/Azure.ResourceManager.Dns/src/Generated/Models/DnsCnameRecordInfo.cs [hunk 3]
```diff
@@ -11,7 +11,7 @@
 namespace Azure.ResourceManager.Dns.Models
 {
     /// <summary> A CNAME record. </summary>
-    internal partial class DnsCnameRecordInfo
+    public partial class DnsCnameRecordInfo
```
Comment (live1206) [inline sdk/dns/Azure.ResourceManager.Dns/src/Generated/Models/DnsCnameRecordInfo.cs:14]: why public?

File: sdk/dns/Azure.ResourceManager.Dns/src/Customization/DnsRecordData.cs [hunk 4]
```diff
@@ -92,7 +93,7 @@ public DnsRecordData()
...
+        public DnsRecordData(ResourceIdentifier id, string name, ResourceType resourceType, ResourceManager.Models.SystemData systemData, ETag? etag, IDictionary<string, string> metadata, long? ttl, string fqdn, string provisioningState, WritableSubResource targetResource, WritableSubResource trafficManagementProfile, IList<DnsARecordInfo> aRecords, IList<DnsAaaaRecordInfo> aaaaRecords, IList<DnsMXRecordInfo> mxRecords, IList<DnsNSRecordInfo> nsRecords, IList<DnsPtrRecordInfo> ptrRecords, IList<DnsSrvRecordInfo> srvRecords, IList<DnsTxtRecordInfo> txtRecords, DnsCnameRecordInfo cnameRecordInfo, DnsSoaRecordInfo soaRecordInfo, IList<DnsCaaRecordInfo> caaRecords, IList<DnsDSRecordInfo> dsRecords, IList<DnsTlsaRecordInfo> tlsaRecords, IList<DnsNaptrRecordInfo> naptrRecords, IDictionary<string, BinaryData> serializedAdditionalRawData) : base(id, name, resourceType, systemData, etag, metadata, ttl, fqdn, provisioningState, targetResource, trafficManagementProfile, serializedAdditionalRawData)
```
[the remark this answers is not in the document]
Comment (welovej) [reply sdk/dns/Azure.ResourceManager.Dns/src/Customization/DnsRecordData.cs:96]: Fixed

File: sdk/dns/Azure.ResourceManager.Dns/src/Customization/DnsRecordData.cs (hunk 1 above)
Comment (welovej) [reply sdk/dns/Azure.ResourceManager.Dns/src/Customization/DnsRecordData.cs:158]: Convert string values to enums, but only modify a few specific strings.

File: sdk/dns/Azure.ResourceManager.Dns/src/Customization/DnsRecordData.cs [hunk 5]
```diff
@@ -125,7 +126,7 @@ internal DnsRecordData(ResourceIdentifier id, string name, ResourceType resource
         /// <summary> The list of TXT records in the record set. </summary>
         public IList<DnsTxtRecordInfo> DnsTxtRecords { get; }
         /// <summary> The CNAME record in the  record set. </summary>
-        internal DnsCnameRecordInfo DnsCnameRecordInfo { get; set; }
+        public DnsCnameRecordInfo DnsCnameRecordInfo { get; set; }
```
[the remark this answers is not in the document]
Comment (welovej) [reply sdk/dns/Azure.ResourceManager.Dns/src/Customization/DnsRecordData.cs:129]: the DnsCnameRecordInfo  formal parameter in DnsRecordData's facotory method, so it needs to be public

File: sdk/dns/Azure.ResourceManager.Dns/src/Generated/Models/DnsCnameRecordInfo.cs (hunk 3 above)
Comment (welovej) [reply sdk/dns/Azure.ResourceManager.Dns/src/Generated/Models/DnsCnameRecordInfo.cs:14]: a DnsCnameRecordInfo formal parameter is in DnsRecordData's facotory method, so it needs to be public

File: sdk/dns/Azure.ResourceManager.Dns/src/Customization/DnsRecordData.cs (hunk 5 above)
[the remark this answers is not in the document]
Comment (welovej) [reply sdk/dns/Azure.ResourceManager.Dns/src/Customization/DnsRecordData.cs:129]: `DnsCnameRecordInfo` formal parameter is in DnsRecordData's facotory method, so it needs to be public

File: sdk/dns/Azure.ResourceManager.Dns/src/Customization/DnsRecordData.cs [hunk 6]
```diff
@@ -142,11 +143,25 @@ public string Cname
...
+        public DnsRecordType RecordType
+        {
+            get
+            {
+                var resourceTypeString = base.ResourceType.Type.Split('/').Where(part => !string.IsNullOrEmpty(part)).LastOrDefault();
+                resourceTypeString = resourceTypeString == "AAAA" ? "Aaaa" : resourceTypeString;
+                resourceTypeString = resourceTypeString == "CNAME" ? "Cname" : resourceTypeString;
+                resourceTypeString = resourceTypeString == "TLSA" ? "Tlsa" : resourceTypeString;
+                resourceTypeString = resourceTypeString == "NAPTR" ? "Naptr" : resourceTypeString;
+                DnsRecordType recordType = (DnsRecordType)Enum.Parse(typeof(DnsRecordType), resourceTypeString);
```
Comment (ArcturusZhang) [inline sdk/dns/Azure.ResourceManager.Dns/src/Customization/DnsRecordData.cs:162]: how about we just add the ignore casing here?
```suggestion
                DnsRecordType recordType = Enum.Parse<DnsRecordType>(typeof(DnsRecordType), resourceTypeString, true);
```

Comment (welovej) [reply sdk/dns/Azure.ResourceManager.Dns/src/Customization/DnsRecordData.cs:162]: use `DnsRecordType recordType = DnsRecordTypeExtensions.ToDnsRecordType(resourceTypeString);`  could be better

File: sdk/dns/Azure.ResourceManager.Dns/src/Customization/ArmDnsModelFactory.cs [hunk 7]
```diff
@@ -358,5 +358,75 @@ public static DnsTxtRecordData DnsTxtRecordData(ResourceIdentifier id = null, st
...
+                targetResource: targetResource,
+                trafficManagementProfile: trafficManagementProfile,
+                aRecords: aRecords,
+                aaaaRecords: aaaaRecords,
+                mxRecords: mxRecords,
+                nsRecords: nsRecords,
+                ptrRecords: ptrRecords,
+                srvRecords: srvRecords,
+                txtRecords: txtRecords,
+                cnameRecordInfo: null,
```
Comment (live1206) [inline sdk/dns/Azure.ResourceManager.Dns/src/Customization/ArmDnsModelFactory.cs:423]: should this be a new instance with cname?

Comment (welovej) [reply sdk/dns/Azure.ResourceManager.Dns/src/Customization/ArmDnsModelFactory.cs]: Fixed

**Review by live1206 — APPROVED** [review]
